modules/virt-reclaiming-persistent-volumes-for-binding-to-a-new-persistent-volume.adoc

```diff
@@ -2,57 +2,64 @@
 //
 // virt/virtual_machines/virtual_disks/virt-reclaiming-persistent-volumes.adoc
 
-[id="virt-reclaiming-persistent-volumes-for-binding-to-a-new-persistent-volume_{context}"]
+[id="virt-reclaiming-statically-provisioned-persistent-volumes-for-binding-to-a-new-persistent-volume_{context}"]
+= Reclaiming statically provisioned persistent volumes
 
-= Reclaiming persistent volumes for binding to a new persistent volume
-
-Use this procedure to reclaim a statically-provisioned persistent volume for reuse.
+Reclaiming a stactically provisioned persistent volume (PV) for re-use is dependent on the underlying storage.
+This procedure provides a general approach that may need to be customised depending on your storage.
 
 .Procedure
-. Export the PV configuration for future use, and edit the YAML file if needed:
+
+. Ensure that no resources are using the persistent volume claim (PVC):
 +
 ----
-$ oc get pv <pv_name> -o yaml > <file_name>.yaml
+$ oc describe pvc <pvc_name> | grep 'Mounted By:'
 ----
 +
-. Enter the following command to display connection details for the persistent volume
-you want to reclaim. Save the output for future reference.
+Remove any resources that are still using the PVC before continuing.
+
+. Delete the PVC to release the PV:
 +
 ----
-$ oc describe pv <pv_name>
+$ oc delete pvc <pvc_name>
 ----
-. Review the `StorageClass` object. Verify that the Storage Class `reclaimPolicy` is set to `delete`.
-If it is not, create a new `StorageClass` object with `reclaimPolicy` set to `delete`.
 
-. Enter the following command to verify that no other resources
-are using the persistent volume claim:
+. Optional: If you do not have a copy of the PV manifest and want to re-use it, export the PV configuration to a YAML file:
 +
 ----
-$ oc describe pvc <pvc_name>
+$ oc get pv <pv_name> -o yaml > <file_name>.yaml
 ----
-. Delete the persistent volume claim to release the persistent volume:
+
+. Ensure that the reclaim policy of the PV is set to `Delete`:
++
+ oc get pv <pv_name> -o yaml | grep 'persistentVolumeReclaimPolicy'
+
+.. If the 'persistentVolumeReclaimPolicy' is not set to `Delete`, edit the reclaim policy with the following command:
```

Comment:
After spending a bit of time reading about storageclasses, and testing behaviour of PVs inheriting reclaim policy from the storageclass, changing the reclaim policy of the PV seemed more appropriate for the procedure than deleting the storageclass and creating a new one.

Reply:
Interesting
So the check should be for Retain instead of Delete, and this should happen before the PVC deleted?
This also sounds like this procedure is only applicable to statically provisioned PVs. If that's correct, I'll add that to the title to improve findability.

Reply:
Well I am saying when you created the original PVs, they should have been retain not delete due to the above. And yes this only applies to statically provisioned PVs because if there is a provisioner it will know how to delete the PVs properly and none of this document applies.